### ISSUE TITLE
clone() return types should not be 'this'

### ISSUE THREAD

**Issue by jeff-tenhave** [Esri/jsapi-resources#90]
All clone() methods in the typings for 4.4 have the return type `this`.  This means that any implementations of clone() must return 'this', which is wrong.

Example: The typings for Camera.
```
interface Camera extends Accessor, JSONSupport {
    fov: number;
    heading: number;
    position: Point;
    tilt: number;

    clone(): this;
  }
```

You definitely don't want to return `this` in the implementation of clone()

**Comment by dasa** [comment]
The `this` type doesn't mean that the JS code is returning `this`. See "Polymorphic this types" at http://www.typescriptlang.org/docs/handbook/advanced-types.html

**Comment by jeff-tenhave** [comment]
True, but it means that if you make any custom implementation of those interfaces, then clone() must return `this`. Example: You extend Collection and add some extra properties. You override clone() to handle your extra properties, in which case you have to return `this` from clone() which is not valid.

**Comment by jkieboom** [comment]
> You extend Collection and add some extra properties. You override `clone()` to handle your extra properties, in which case you have to return `this` from `clone()` which is not valid.

Should be fixed in 4.5.
